packages/pigeon/lib/objc_generator.dart

```diff
@@ -612,8 +625,13 @@ void _writeHostApiSource(Indent indent, ObjcOptions options, Api api) {
       indent.writeln('NSArray *args = $variable;');
       map3(wholeNumbers.take(func.arguments.length), argNames, func.arguments,
           (int count, String argName, NamedType arg) {
-        final _ObjcPtr argType = _objcTypeForDartType(options.prefix, arg.type);
-        return '${argType.ptr}$argName = GetNullableObjectAtIndex(args, $count);';
+        if (_isEnum(root, arg.type)) {
+          return '${_className(options.prefix, arg.type.baseName)} $argName = [GetNullableObjectAtIndex(args, $count) integerValue];';
```

Comment:
Unless I'm misunderstanding the code this generates, this would convert a null value from dart to whatever enum value zero is.

Reply:
Yep, but we are blocking nullable enums right now.